2_process/src/process_dv_stat_colors.R

```diff
@@ -3,15 +3,20 @@
 #' @param ind_file character file name where the output should be saved
 #' @param dv_stats_ind indicator file for the data.frame of dv_data
 #' @param color_palette list of colors to use for the color ramp (from viz_config.yml)
-process_dv_stat_colors <- function(ind_file, dv_stats_ind, color_palette){
+process_dv_stat_colors <- function(ind_file, dv_stats_ind, color_palette, gage_style){
 
   dv_stats <- readRDS(scipiper::sc_retrieve(dv_stats_ind, remake_file = '2_process.yml'))
-  col_fun <- colorRamp(color_palette)
-
-  # just removing NA percentiles for now
+  col_fun <- colorRamp(color_palette$with_percentile)
   dv_stats_with_color <- dv_stats %>%
-    filter(!is.na(per)) %>%
-    mutate(color = rgb(col_fun(per), maxColorValue = 255)) # don't know how necessary maxColorValue is
+    mutate(shape = ifelse(is.na(per), no = gage_style$with_percentile$pch,
+                         yes = gage_style$no_percentile$pch),
+          size = ifelse(is.na(per), no = gage_style$with_percentile$cex,
+                        yes = gage_style$no_percentile$cex),
+          color = NA)
+  per_na <- is.na(dv_stats$per)
+  #this was not working with ifelse inside mutate, still NAs getting into rgb/col_fun
```

Comment:
I wonder if you needed to do `rowwise()` or something?

Reply:
Using `rowwise` made it not error, but the result was still wrong 😕

Reply:
well this works with your base R solution below, so moving on!